### PR TITLE
[v14] Add Posthog events for discovered Kubernetes Apps

### PR DESCRIPTION
**Pull request by tigrato**
Backport #32289 to branch/v14

### CODE REVIEW

**Comment by tigrato** [comment]
On hold until the cloud counterpart lands in prod env

**Review by AntonAM — APPROVED** [review]

**Review by ravicious — APPROVED** [review]

**Pull request merged**
